### PR TITLE
Fixes #1512, don't save edits with no clips

### LINKED ISSUE #1512: Editor will produce an invalid SMIL file if user tries to delete whole video
### To Reproduce:

Go to stable.opencast.org edit the dual-stream-demo

1. Create a segment and mark it for deletion.
2. Stretch the segment to the end of the timeline.
3. Save and process
4. The workflow will fail on the editor WOH with:

```
org.opencastproject.workflow.api.WorkflowOperationException: org.opencastproject.smil.api.SmilException: Smil does not define any elements
	at org.opencastproject.workflow.handler.videoeditor.VideoEditorWorkflowOperationHandler.resume(VideoEditorWorkflowOperationHandler.java:470)
	at org.opencastproject.workflow.handler.videoeditor.VideoEditorWorkflowOperationHandler.start(VideoEditorWorkflowOperationHandler.java:31
```

### Expected/Desired behaviour

The editor should not allow the user remove the whole video. Instead it should inform the user to delete video via the Opencast Admin UI or simply leave it unpublished.

### PR DESCRIPTION
**Pull request by JamesUoM**
Before allowing the edits to be saved/processed, check that there isn't a single segment that deletes the whole video. Otherwise Opencast will create a SMIL file with not clips and the editor WOH will fail.

### CODE REVIEW

File: src/i18n/locales/en-US.json [hunk 1]
```diff
@@ -76,7 +76,9 @@
       "success-tooltip-aria": "Saved successfully",
       "saveArea-tooltip": "Save Area",
       "confirm-success": "Okay",
-      "cancel-save": "Don't save"
+      "cancel-save": "Don't save",
+      "invalid-headline-text": "Invalid Edits",
+      "invalid-text": "The segments do not create a valid video. Either change or discard your edits if you wish to proceed. If you wanted to delete this video use the Opencast Admin UI. Contact an adminstrator for further help."
```
Comment (Arnei) [inline src/i18n/locales/en-US.json:81]: We should avoid mentioning the Admin UI. The editor can be accessed from other frontends as well, like Tobira or LMS plugins. User may not know what the "Opencast Admin UI" is, nor should they need to.

**Review by Arnei — CHANGES_REQUESTED** [review]
I've found some things that I believe should be addressed before merging.

I'm not happy with how we present the user with the problem. Simply replacing the usual save text with a warning is not obvious at first glance and does not really make clear that it is supposed to be warning. Maybe keeping the usual text and instead adding a warning box would be better? (Could use the error boxes from appkit for that *hint hint* https://github.com/opencast/opencast-editor/pull/1392). I would also prefer to keep rendering the save button, but disable it.

**Comment by JamesUoM** [comment]
> Maybe keeping the usual text and instead adding a warning box would be better? (Could use the error boxes from appkit for that _hint hint_ #1392). I would also prefer to keep rendering the save button, but disable it.

Agree with using the error boxes. But disabling buttons is not meant to be good accessible UX design - also I did try but as they are not form elements, disabling doesn't work. The `aria-disabled` will tell screen readers that it's disabled but doesn't stop it functioning.

File: src/redux/videoSlice.ts [hunk 2]
```diff
@@ -184,6 +186,15 @@ const videoSlice = createSlice({
       updateCurrentlyAt(state, jumpTarget);
       state.jumpTriggered = true;
     },
+    validateSegments: state => {
+      // Test if whole video has been deleted
+      if (state.segments.length === 1 && state.segments[0].deleted && state.segments[0].start === 0 &&
```
[the remark this answers is not in the document]
Comment (JamesUoM) [reply src/redux/videoSlice.ts:191]: yes, I considered that but thought it a bit of an edge case. I will add some additional tests though.

**Comment by Arnei** [comment]
My concerns have been addressed, thanks! 

> But disabling buttons is not meant to be good accessible UX design

I disagree with this as a general statement. While disabling a button *alone* is indeed not good UX, I do believe it can be valuable if there are additional measures taken that make it clear to the user why it is disabled. It then helps making clear that there indeed would be a button here that they could click, instead of having the button suddenly appear.

But apparently whether disabling buttons is a good or bad thing is a whole big discussion that I do not find worth getting into since the PR looks fine to me.

> also I did try but as they are not form elements, disabling doesn't work. The aria-disabled will tell screen readers that it's disabled but doesn't stop it functioning.

Oh good to know. That's an issue with ProtoButton from appkit, right? I'll open an issue over there.

**Review by Arnei — APPROVED** [review]

**Comment by gregorydlogan** [comment]
~If/when you fix this @JamesUoM, remember to change the branch target to `r/17.x` if you want this in a 17.x release!~

Nevermind, turns out I can fix that myself.

**Comment by gregorydlogan** [comment]
@JamesUoM this, IMO, should still be fixed.  But maybe in 18/19 since we probably don't want to change this in legacy at this point.

**Comment by KatrinIhler** [comment]
@gregorydlogan Is this important enough that we should fix the conflicts and merge to 18? Otherwise I'm gonna close this.

**Comment by gregorydlogan** [comment]
I mean, this kinda feels like something that should be allowed to happen, but it's also something that you're really only going to see in users who are trying to be difficult.

Can close, can refile if/when someone wants to get this in.
